planning/mission_planner/src/rerouting_static_obstacle/rerouting_static_obstacle.cpp

```diff
@@ -0,0 +1,203 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rerouting_static_obstacle.hpp"
+
+#include <lanelet2_extension/utility/message_conversion.hpp>
+#include <lanelet2_extension/utility/query.hpp>
+#include <lanelet2_extension/utility/utilities.hpp>
+
+#include <limits>
+#include <memory>
+
+namespace mission_planner
+{
+ReroutingStaticObstacle::ReroutingStaticObstacle(const rclcpp::NodeOptions & node_options)
+: Node("rerouting_static_obstacle", node_options)
+{
+  using std::placeholders::_1;
+
+  sub_odom_ = this->create_subscription<nav_msgs::msg::Odometry>(
+    "input/odometry", rclcpp::QoS(1), std::bind(&ReroutingStaticObstacle::on_odom, this, _1));
+
+  sub_route_ = create_subscription<autoware_planning_msgs::msg::LaneletRoute>(
+    "input/route", rclcpp::QoS{1}.transient_local(),
+    std::bind(&ReroutingStaticObstacle::route_callback, this, std::placeholders::_1));
+
+  map_subscriber_ = create_subscription<HADMapBin>(
+    "input/vector_map", rclcpp::QoS{10}.transient_local(),
+    std::bind(&ReroutingStaticObstacle::map_callback, this, std::placeholders::_1));
+
+  sub_trigger_ = this->create_subscription<geometry_msgs::msg::PointStamped>(
+    "input/reroute_point", rclcpp::QoS(1),
+    std::bind(&ReroutingStaticObstacle::on_trigger, this, _1));
+
+  const auto adaptor = component_interface_utils::NodeAdaptor(this);
+  adaptor.init_cli(cli_change_route_);
+}
+
+void ReroutingStaticObstacle::on_odom(const nav_msgs::msg::Odometry::SharedPtr msg)
+{
+  current_pose_ = msg->pose.pose;
+}
+
+void ReroutingStaticObstacle::route_callback(
+  const autoware_planning_msgs::msg::LaneletRoute::ConstSharedPtr msg)
+{
+  goal_pose_ = msg->goal_pose;
+}
+
+void ReroutingStaticObstacle::map_callback(const HADMapBin::ConstSharedPtr msg)
+{
+  route_handler_.setMap(*msg);
+  lanelet_map_ptr_ = std::make_shared<lanelet::LaneletMap>();
+  lanelet::utils::conversion::fromBinMsg(
+    *msg, lanelet_map_ptr_, &traffic_rules_ptr_, &routing_graph_ptr_);
+  lanelet::ConstLanelets all_lanelets = lanelet::utils::query::laneletLayer(lanelet_map_ptr_);
+  road_lanelets_ = lanelet::utils::query::roadLanelets(all_lanelets);
+}
+
+void ReroutingStaticObstacle::on_trigger(const geometry_msgs::msg::PointStamped::SharedPtr msg)
+{
+  geometry_msgs::msg::Point selected_point{msg->point};
+
+  lanelet::ConstLanelets selected_point_lanelets;
+
+  auto selected_point_lanelet_found =
+    get_selected_point_lanelets(selected_point, selected_point_lanelets);
```

Comment:
```suggestion
  const auto selected_point_lanelet_found =
    get_selected_point_lanelets(selected_point, selected_point_lanelets);
```

please add const and for other parts